### ISSUE TITLE
[JENKINS-64614] clang-tidy tool reports warnings generated by gcc

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/analysis-model#1378]
<p>A bit related to <a href="https://issue-redirect.jenkins.io/issue/64612" title="clang-tidy warnigns are reported as gcc warnings" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/64612">JENKINS-64612</a>"><del>JENKINS-64612</del></a>. The clang-tidy tool reports warnings generated by gcc.</p>

<p>This is a bigger problem than it may seem. You could argue "well, just put them in different files for different tools to parse", but a common way to run clang-tidy is via cmake as </p>

```
cmake -DCMAKE_CXX_CLANG_TIDY=clang-tidy .. && cmake --build .
```
<p>Meaning both the clang-tidy and gcc warnings are generated as a result of the same command being executed. This can't be separated (<a href="https://discourse.cmake.org/t/run-clang-tidy-only/2537" class="external-link" target="_blank" rel="nofollow noopener">https://discourse.cmake.org/t/run-clang-tidy-only/2537</a>), and has benefits over other ways of running clang-tidy (only check the files that have changed in incremental builds).</p>

<p>In the same way <a href="https://issue-redirect.jenkins.io/issue/64612" title="clang-tidy warnigns are reported as gcc warnings" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/64612">JENKINS-64612</a>"><del>JENKINS-64612</del></a> argues for only including warnings with a "-W" in them, here I would like to argue that the clang-tidy tool should filter-out the warnings with a "-W" in them.</p>

<p>In fact, as of clang-tidy 11.0.0, the only categories clang-tidy reports are:</p>

```
abseil-*
android-*
boost-*
bugprone-*
cert-*
clang-analyzer-*
cppcoreguidelines-*
darwin-*
fuchsia-*
google-*
hicpp-*
linuxkernel-*
llvm-*
llvmlibc-*
misc-*
modernize-*
mpi-*
objc-*
openmp-*
performance-*
portability-*
readability-*
zircon-*
```
<p>(none of them starting with "-W").</p>

<p> </p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=reddwarf94">reddwarf94</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-64614" target="_blank">clang-tidy tool reports warnings generated by gcc</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=akash_manna">akash_manna</a>
<li><b>status</b>: Fixed but Unreleased
<li><b>priority</b>: Minor
<li><b>component(s)</b>: analysis-model
<li><b>label(s)</b>: hacktoberfest, help-wanted
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2025-11-09T21:26:30+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 1
<li><b>imported</b>: 20260108-221235
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>A bit related to <a href="https://issues.jenkins.io/browse/JENKINS-64612" title="clang-tidy warnigns are reported as gcc warnings" class="issue-link" data-issue-key="JENKINS-64612"><del>JENKINS-64612</del></a>. The clang-tidy tool reports warnings generated by gcc.</p>

<p>This is a bigger problem than it may seem. You could argue "well, just put them in different files for different tools to parse", but a common way to run clang-tidy is via cmake as </p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>cmake -DCMAKE_CXX_CLANG_TIDY=clang-tidy .. &amp;&amp; cmake --build .</pre>
</div></div>
<p>Meaning both the clang-tidy and gcc warnings are generated as a result of the same command being executed. This can't be separated (<a href="https://discourse.cmake.org/t/run-clang-tidy-only/2537" class="external-link" target="_blank" rel="nofollow noopener">https://discourse.cmake.org/t/run-clang-tidy-only/2537</a>), and has benefits over other ways of running clang-tidy (only check the files that have changed in incremental builds).</p>

<p>In the same way <a href="https://issues.jenkins.io/browse/JENKINS-64612" title="clang-tidy warnigns are reported as gcc warnings" class="issue-link" data-issue-key="JENKINS-64612"><del>JENKINS-64612</del></a> argues for only including warnings with a "-W" in them, here I would like to argue that the clang-tidy tool should filter-out the warnings with a "-W" in them.</p>

<p>In fact, as of clang-tidy 11.0.0, the only categories clang-tidy reports are:</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>abseil-*
android-*
boost-*
bugprone-*
cert-*
clang-analyzer-*
cppcoreguidelines-*
darwin-*
fuchsia-*
google-*
hicpp-*
linuxkernel-*
llvm-*
llvmlibc-*
misc-*
modernize-*
mpi-*
objc-*
openmp-*
performance-*
portability-*
readability-*
zircon-*</pre>
</div></div>
<p>(none of them starting with "-W").</p>

<p> </p></pre>
</details>
</details>
